notebooks/1_intelligent_agents_fall_2021/index.ipynb

```diff
@@ -0,0 +1,225 @@
+    "we can describe an agent's behavior by the agent function.  \n",
+    "<b>agent function</b> maps any given percepts sequence to an action. but how does the agent know what sequence it must choose? we will try to answer this question using a simple example."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Vacuum world agent example\n",
+    "<img src=\"./images/vacumm_world.png\" width=\"500\" style=\"margin-left: auto;margin-right: auto;\"/>\n",
+    "Imagine an environment that only has two rooms (room A & room B). our agent is a vacuum cleaner. it can perceive whether the room it is currently in, is dirty or not. it can also move between the rooms and suck up dirt (the actuators of the agent).  \n",
+    "Imagine that we only care about both rooms being clean, a simple way we can implement an agent that guarantees this is for our agent to constantly move between the rooms and suck up dirt in each one. Now imagine we also want to minimize the amount of energy the vacuum cleaner uses. In order to do that we can stop the agent for a limited amount of time if both rooms are clean. as you can see we implemented two completely different agents for these two problems and that's because we used two different performance measures."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Rational agents and performance measure\n",
+    "a <b>rational</b> agent choose the set of action in order to maximize its performance. agents use a performance measure to evaluate the desirability of any given sequence. In other words, an agent will choose the action (or a sequence of them) that maximize the expected value of its performance measure."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Rationality vs perfection\n",
+    "Keep in mind that rationality is distinct from omniscience. an omniscience agent knows the actual outcome of its actions but in reality, an agent only knows the expected outcome of its action.\n",
+    "#### Autonomy\n",
+    "a rational agent should be autonomous meaning it mustn't only rely on the prior knowledge of its designer and must learn to compensate for partial or incorrect prior knowledge. In other words, rational agents should learn from experience. for example, in the vacuum world our agent could start to learn when the rooms usually get dirty based on its experience."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Task environment (PEAS)\n",
+    "we have already talked about performance measure, task environment, actuators and sensors. we group all these under the heading of the <b>Task enviroment </b> and we abbreviate it as <b>PEAS</b>(<b>P</b>erformance measure, <b>E</b>nviroment, <b>A</b>ctuators, <b>S</b>ensors). When designing an agent our first step should be specifying the task enviroment.\n",
+    "#### Types of environment\n",
```

Comment:
This part is too brief. You should explain them in much more detail using examples. The reader should gain more information when he/she studies your markdown in comparison with slides!